src/core/stream/initial_seek_and_play.ts

```diff
@@ -89,33 +143,82 @@ export default function seekAndLoadOnMediaEvents(
   mustAutoPlay : boolean
 ) : {
   seek$ : Observable<void>;
-  load$ : Observable<"autoplay-blocked"|"autoplay"|"loaded">;
+  load$ : Observable<{
+    evt: "autoplay-blocked"|"autoplay"|"loaded";
+    playPauseEvents$: Observable<boolean>;
+  }>;
 } {
+  let playPauseEvents$: Observable<boolean>;
   const seek$ = doInitialSeek(mediaElement, startTime);
-
-  const load$ = canPlay(mediaElement).pipe(
+  const load$ = observableConcat(
+    seek$.pipe(ignoreElements()),
+    canPlay(mediaElement)
+  ).pipe(
 
     tap(() => log.info("Stream: canplay event")),
 
     mergeMap(() => {
-      if (mustAutoPlay) {
-        return play$(mediaElement).pipe(
-          mapTo("autoplay" as "autoplay"),
-          catchError((error) => {
-            if (error.name === "NotAllowedError") {
-              // auto-play was probably prevented.
-              log.warn("Stream: Media element can't play." +
-                " It may be due to browser auto-play policies.");
-              return observableOf("autoplay-blocked" as "autoplay-blocked");
-            } else {
-              throw error;
+
+      const initialMediaDuration = mediaElement.duration;
+      let autoPlayBlocked = false;
+
+      playPauseEvents$ = onPlayPause$(mediaElement).pipe(
+        map((x) => (x && x.type === "play") ? true : false),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Simplify this expression. ([typescript:S1125](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS1125))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=385&issues=AWdAABQ2Gn-j85A9-WKY&open=AWdAABQ2Gn-j85A9-WKY)

[ISSUE_KEY]: AWdAABQ2Gn-j85A9-WKY